perfdash/Dockerfile

```diff
@@ -12,8 +12,22 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM debian:jessie
-RUN apt-get update
-RUN apt-get install -y -qq ca-certificates
-ADD perfdash /perfdash
-ADD www /www
+ARG GOLANG_VERSION=1.13
+FROM golang:${GOLANG_VERSION} as builder
+
+WORKDIR /workspace
+
+# Copy the Go Modules manifests
+COPY go.sum go.sum
+COPY go.mod go.mod
+RUN go mod download
+
+COPY *.go .
```

Comment:
I had no issues running this command.

```shell
PASS
ok  	k8s.io/perf-tests/perfdash	0.012s
go build -a -installsuffix cgo -ldflags '-w' -o perfdash
docker build --pull -t gcr.io/k8s-testimages/perfdash:2.35 .
[+] Building 1.2s (16/16) FINISHED                                                                                                                                                                                                                                                                                                                                                                                                         
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                                                                                                                                                                                  0.0s
 => => transferring dockerfile: 37B                                                                                                                                                                                                                                                                                                                                                                                                   0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                                                                                                                                                                                     0.1s
 => => transferring context: 2B                                                                                                                                                                                                                                                                                                                                                                                                       0.0s
 => [internal] load metadata for gcr.io/distroless/base:nonroot                                                                                                                                                                                                                                                                                                                                                                       0.8s
 => [internal] load metadata for docker.io/library/golang:1.13                                                                                                                                                                                                                                                                                                                                                                        0.9s
 => [builder 1/7] FROM docker.io/library/golang:1.13@sha256:8ebb6d5a48deef738381b56b1d4cd33d99a5d608e0d03c5fe8dfa3f68d41a1f8                                                                                                                                                                                                                                                                                                          0.0s
 => [internal] load build context                                                                                                                                                                                                                                                                                                                                                                                                     0.1s
 => => transferring context: 703B                                                                                                                                                                                                                                                                                                                                                                                                     0.0s
 => [stage-1 1/3] FROM gcr.io/distroless/base:nonroot@sha256:ccbc79c4fc35b92709d3987315cdb9e20b6e742546af7a7db10024641aa60572                                                                                                                                                                                                                                                                                                         0.0s
 => CACHED [stage-1 2/3] COPY www/ /www/                                                                                                                                                                                                                                                                                                                                                                                              0.0s
 => CACHED [builder 2/7] WORKDIR /workspace                                                                                                                                                                                                                                                                                                                                                                                           0.0s
 => CACHED [builder 3/7] COPY go.sum go.sum                                                                                                                                                                                                                                                                                                                                                                                           0.0s
 => CACHED [builder 4/7] COPY go.mod go.mod                                                                                                                                                                                                                                                                                                                                                                                           0.0s
 => CACHED [builder 5/7] RUN go mod download                                                                                                                                                                                                                                                                                                                                                                                          0.0s
 => CACHED [builder 6/7] COPY *.go .                                                                                                                                                                                                                                                                                                                                                                                                  0.0s
 => CACHED [builder 7/7] RUN GO111MODULE=on go build -a -installsuffix cgo -ldflags '-w' -o $PWD/perfdash                                                                                                                                                                                                                                                                                                                             0.0s
 => CACHED [stage-1 3/3] COPY --from=builder /workspace/perfdash /                                                                                                                                                                                                                                                                                                                                                                    0.0s
 => exporting to image                                                                                                                                                                                                                                                                                                                                                                                                                0.0s
 => => exporting layers                                                                                                                                                                                                                                                                                                                                                                                                               0.0s
 => => writing image sha256:5a77bd57f5011f0e6896bf1fa35437c23d6e207996d20854c54bcad94f556515                                                                                                                                                                                                                                                                                                                                          0.0s
 => => naming to gcr.io/k8s-testimages/perfdash:2.35 
```

It's possible it's related to the version of Docker used. Mine: 

```shell
 docker version
Client:
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.16.6
 Git commit:        v20.10.7
 Built:             Thu Jan  1 00:00:00 1970
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true
```

Reply:
Pushed changes to fix the make error.

Reply:
Interesting, it still fails on my side despite having quite similar version:

```
Client:
 Version:           20.10.2
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        2291f61
 Built:             Mon Dec 28 16:17:34 2020
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true
```

It finishes with:
```
➜  perfdash git:(9fe19061) make container
go test
PASS
ok  	k8s.io/perf-tests/perfdash	0.079s
go build -a -installsuffix cgo -ldflags '-w' -o perfdash
docker build --pull -t gcr.io/k8s-testimages/perfdash:2.35 .
Sending build context to Docker daemon  33.62MB
Step 1/12 : ARG GOLANG_VERSION=1.13
Step 2/12 : FROM golang:${GOLANG_VERSION} as builder
1.13: Pulling from library/golang
Digest: sha256:8ebb6d5a48deef738381b56b1d4cd33d99a5d608e0d03c5fe8dfa3f68d41a1f8
Status: Image is up to date for golang:1.13
 ---> d6f3656320fe
Step 3/12 : WORKDIR /workspace
 ---> Using cache
 ---> f75b91b08687
Step 4/12 : COPY go.sum go.sum
 ---> Using cache
 ---> c3d799b7a1ff
Step 5/12 : COPY go.mod go.mod
 ---> Using cache
 ---> 153cb78cc924
Step 6/12 : RUN go mod download
 ---> Using cache
 ---> 56ca1fadb02c
Step 7/12 : COPY *.go /workspace
When using COPY with more than one source file, the destination must be a directory and end with a /
make: *** [Makefile:24: container] Error 1
```

Reply:
I'm not sure why it fails on my workstation while it works on yours, but can't we simply add a a trailing slash as suggested in the error message?

Something like
```
COPY *.go /workspace/
```

should work in all environments

Reply:
Done. 
I'll also suggest to move the auto-(build and push) system based on CloudBuild like we do with the k8s images: https://github.com/kubernetes/k8s.io/tree/main/k8s.gcr.io. 
 It could help avoid discrepancy between different environments.